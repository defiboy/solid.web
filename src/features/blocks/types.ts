import { Transaction } from "features/transactions";

export interface BlockHeader {
    number: number
    hash: string
    parentHash: string
    nonce: string
    sha3Uncles: string
    logsBloom: string
    transactionRoot: string
    stateRoot: string
    receiptRoot: string
    miner: string
    extraData: string
    gasLimit: number
    gasUsed: number
    timestamp: number | string
}

export interface Block extends BlockHeader {
    transactions: Transaction[];
    size: number
    difficulty: number
    totalDifficulty: number
    uncles: string[];
}