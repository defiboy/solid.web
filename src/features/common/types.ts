export enum Status { // TODO, with sagas I think it won't apply
    InProgress = 'InProgress',
    Completed = 'Completed',
    Synchronizing = 'Synchronizing', // this won't apply for connection modal
    Failed = 'Failed',
    NotStarted = 'NotStarted',
    Started = 'Started'
}

export interface GenericResponse<T> {
    total: number;
    limit: number;
    skip: number;
    data: T;
}

export interface GenericArrayResponse<T> {
    total: number;
    limit: number;
    skip: number;
    data: T[];
}
