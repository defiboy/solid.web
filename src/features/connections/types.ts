import { Connection } from '@solid-explorer/types'

export interface ConnectionItem extends Connection {
  type: string
}
