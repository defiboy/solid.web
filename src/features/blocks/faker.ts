import { Block } from "./types"

export const buildFakeBlock = (): Block => {
    return {
        difficulty: 54831103175033,
        extraData: "0xd783010406844765746887676f312e342e32856c696e7578",
        gasLimit: 4712388,
        gasUsed: 3059685,
        hash: "0xec4cd4f70ab16718ada2795b0b24f2be6498f4fca3e0b6b79e045933814f56e7",
        logsBloom: "0x00000000000000000400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000002000000000000000000000000000000000000000000000000002000000000000000000000000000000000",
        miner: "0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5",
        nonce: "0x8b119d486f87adf5",
        number: 1757201,
        parentHash: "0xd4bd6a69b51d768950f683151e009410b74403f8dc2780bf03697e3a6e4919c7",
        receiptRoot: "0x23ed6181c74edbcd2b29aa317dc84d320788a2c98b818b5fc49c475ef1d5164a",
        sha3Uncles: "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
        size: 10736,
        stateRoot: "0x857cd117a16db6661eee8256702aa6879ecceccdb05a13637bad147760762547",
        timestamp: 1466691328,
        totalDifficulty: 30162772625997950877,
        transactions: ["0xd5875cdc5dea095e713b2409d585bd713d7f67bf1f0a0143293c00dc3db187da", "0xb7f5a83267df37248b44a7204b974a89d5ee870f00e91b6b2858b0088f8ada68", "0xc0fdbd563a0e26966e2f9803729f22bcaf29f0e73df158ccb0868058ce103ef2", "0xb294881e48ed00fe449b9604a29bea78db19ff1dee845f0af26eb75546a0a2f0", "0x0dda4a0efb859327c4755c9610bfafd921647542a60cd4d7fab37a2a1ac74b01", "0xbec6c60e5092138f52d0437c67528459d6a727332022b3686f1bc0df171a4c92", "0x5a9941189e91c5b1a68f8efd7a63b4aa28cb0fc72365d6bd4b2fabedd70da984", "0x13b66d8a1ce8eec4492e070037670c993477371c895e43d6dc498bc5d36ef726", "0x3db1158457a8659d9b249b06ccac059358eb110f9f136f229f94afb76144efdb", "0x15aa6a7e18237f6011bf5660c7c7fcdad57b00e2bd1b83179ed75c3a4d0521d3", "0xec830f03d41367e31d27578f1d981f45377accf0526ed9a9bc691e83b64694c2", "0x1d975af035632721617103e05bc0a4f035d9885ba8d279915dd9dee182a6c794", "0xbe0fc22dd5c7d5330cd5987ee63080d272c5802c7fee508602daf0220f3617df", "0x8dc9ef3427b16c240e6d4e2f4e428db575d5b55a89d22a392f4a0efa43110036", "0xd64b8106cbf8cd61288b765504125f68f9a74666a7aa2187bf83ac2a97cb6efd", "0x000c56bb6b29a1230cbceb3dfc165e15c040be8e5fcf1cb0c62397525bf528c1", "0x096f1b0ed9b7741582ee065bc9c725df0dbd5081fce37f1758c78c96028faa92", "0xed8af8503bfa73b1f5f92135712ef09c71db6e8e5b1a83fc2a5cd6e0fec5639a", "0x1e881a3643f099d84f34c3d5d0aa155cf737ea5348902d2675dec054e803cd11", "0x0289d236f7f9d14b82bd174cf317533566314231cb64c79ef1ec5532638b737b", "0xfa05d618ff5946eec57cb4770daa10f1751fed90a274688eccff2cb82c7365b5", "0x65fd378aec19b1e5302a14fac44077f4df13d5859f3fd27c86931c8ca0e5d0bd"],
        transactionRoot: "0xfe3316b06b1fad15e6e1c0e06ad2043213f36518f122ad6dff0293fe57d26d5f",
        uncles: []
    }
}

export const buildFakeBlocks = (): Block[] => {
    return [
        buildFakeBlock(),
        {
            difficulty: 54831103175033,
            extraData: "0xd783010406844765746887676f312e342e32856c696e7578",
            gasLimit: 4712388,
            gasUsed: 646893,
            hash: "0xdac2431708c0698b66a3a4a3ad4ba44b268df402b9350a909175d6ce6ca9004d",
            logsBloom: "0x00000000000000000400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000002000000000000000000000000000000000000000000000000002000000000000000000000000000000000",
            miner: "0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5",
            nonce: "0x8b119d486f87adf5",
            number: 646893,
            parentHash: "0xdac2431708c0698b66a3a4a3ad4ba44b268df402b9350a909175d6ce6ca9004e",
            receiptRoot: "0x23ed6181c74edbcd2b29aa317dc84d320788a2c98b818b5fc49c475ef1d5164a",
            sha3Uncles: "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
            size: 10736,
            stateRoot: "0x857cd117a16db6661eee8256702aa6879ecceccdb05a13637bad147760762547",
            timestamp: 1466691328,
            totalDifficulty: 30162772625997950877,
            transactions: ["0xd5875cdc5dea095e713b2409d585bd713d7f67bf1f0a0143293c00dc3db187da", "0xb7f5a83267df37248b44a7204b974a89d5ee870f00e91b6b2858b0088f8ada68", "0xc0fdbd563a0e26966e2f9803729f22bcaf29f0e73df158ccb0868058ce103ef2", "0xb294881e48ed00fe449b9604a29bea78db19ff1dee845f0af26eb75546a0a2f0", "0x0dda4a0efb859327c4755c9610bfafd921647542a60cd4d7fab37a2a1ac74b01", "0xbec6c60e5092138f52d0437c67528459d6a727332022b3686f1bc0df171a4c92", "0x5a9941189e91c5b1a68f8efd7a63b4aa28cb0fc72365d6bd4b2fabedd70da984", "0x13b66d8a1ce8eec4492e070037670c993477371c895e43d6dc498bc5d36ef726", "0x3db1158457a8659d9b249b06ccac059358eb110f9f136f229f94afb76144efdb", "0x15aa6a7e18237f6011bf5660c7c7fcdad57b00e2bd1b83179ed75c3a4d0521d3", "0xec830f03d41367e31d27578f1d981f45377accf0526ed9a9bc691e83b64694c2", "0x1d975af035632721617103e05bc0a4f035d9885ba8d279915dd9dee182a6c794", "0xbe0fc22dd5c7d5330cd5987ee63080d272c5802c7fee508602daf0220f3617df", "0x8dc9ef3427b16c240e6d4e2f4e428db575d5b55a89d22a392f4a0efa43110036", "0xd64b8106cbf8cd61288b765504125f68f9a74666a7aa2187bf83ac2a97cb6efd", "0x000c56bb6b29a1230cbceb3dfc165e15c040be8e5fcf1cb0c62397525bf528c1", "0x096f1b0ed9b7741582ee065bc9c725df0dbd5081fce37f1758c78c96028faa92", "0xed8af8503bfa73b1f5f92135712ef09c71db6e8e5b1a83fc2a5cd6e0fec5639a", "0x1e881a3643f099d84f34c3d5d0aa155cf737ea5348902d2675dec054e803cd11", "0x0289d236f7f9d14b82bd174cf317533566314231cb64c79ef1ec5532638b737b", "0xfa05d618ff5946eec57cb4770daa10f1751fed90a274688eccff2cb82c7365b5", "0x65fd378aec19b1e5302a14fac44077f4df13d5859f3fd27c86931c8ca0e5d0bd"],
            transactionRoot: "0xfe3316b06b1fad15e6e1c0e06ad2043213f36518f122ad6dff0293fe57d26d5f",
            uncles: []
        }
    ]
}