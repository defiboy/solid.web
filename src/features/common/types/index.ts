export enum Status { // TODO, with sagas I think it won't apply
    InProgress = 'InProgress',
    Completed = 'Completed',
    Synchronizing = 'Synchronizing', // this won't apply for connection modal
    Failed = 'Failed',
    NotStarted = 'NotStarted',
    Started = 'Started'
}