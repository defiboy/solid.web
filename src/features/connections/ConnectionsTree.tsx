import React from 'react'

import { Action, ActionCreator } from 'redux'

import { Icon } from 'antd'

import { Connection } from './types'
import { MenuStyled, MenuItemStyled, SidebarHeader, SidebarTitle, SidebarHeaderButtons, DirectoryTreeStyled, TreeNodeStyled } from './components';
// Contract
// TODO: I need to create contract instances and render per connection

interface Props {
  connections: Connection[]
  // contracts: Contract[]
  onNewConnectionClick: ActionCreator<Action>
  onContractSelected: ActionCreator<Action>
}

interface State {
  rightClickNodeTreeItem: any
  selectedKeys: any[]
}

export class ConnectionsTree extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      rightClickNodeTreeItem: {},
      selectedKeys: []
    }
  }

  onSelect = (selectedKeys: any, info: any) => {
    // TODO: Show contract on editor when clicked
    // const contractToShow = this.props.contracts.find(item => {
    //   return item._id === selectedKeys[0]
    // })
    // this.props.onContractSelected(contractToShow)
    this.setState({
      selectedKeys
    })
  }

  rightClickOnTree = ({ event, node }: any) => {
    const id = node.props.eventKey
    this.setState({
      rightClickNodeTreeItem: {
        pageX: event.pageX,
        pageY: event.pageY,
        id,
        categoryName: node.props.eventKey
      },
      selectedKeys: [id]
    })
  }

  getNodeTreeRightClickMenu = () => {
    const { pageX, pageY } = { ...this.state.rightClickNodeTreeItem } as any
    if (!pageX || !pageY) {
      return <div />
    }
    const menu = (
      <MenuStyled style={{ position: 'absolute', left: `${pageX}px`, top: `${pageY}px` }}>
        <MenuItemStyled key="1">Deploy</MenuItemStyled>
        <MenuItemStyled key="2">Open Console</MenuItemStyled>
      </MenuStyled>
    )
    return menu
  }

  render() {
    const { connections, onNewConnectionClick } = this.props
    return (
      <div style={{ overflow: 'scroll', height: '100%' }}>
        {this.getNodeTreeRightClickMenu()}
        <SidebarHeader>
          <SidebarTitle>Connections</SidebarTitle>
          <SidebarHeaderButtons>
            <Icon type="plus" style={{ color: 'white', paddingRight: '0.5em' }} onClick={onNewConnectionClick} />
            <Icon type="down" style={{ color: 'white' }} />
          </SidebarHeaderButtons>
        </SidebarHeader>
        {connections && connections.length > 0 && (
          <DirectoryTreeStyled
            onSelect={this.onSelect}
            multiple={true}
            onRightClick={this.rightClickOnTree}
            selectedKeys={this.state.selectedKeys}
            defaultExpandAll={true}
            style={{ color: 'white' }}
          >
            {connections.map(item => {
              return (
                <TreeNodeStyled
                  icon={<Icon type="database" />}
                  title={item.name}
                  key={item.url}
                  style={{ color: 'white' }}
                >
                  <TreeNodeStyled title="Contract Instances" key="0-0-0" style={{ color: 'white' }}>
                    {/* {contracts &&
                      contracts.length > 0 &&
                      contracts.map((contract: Contract) => {
                        return <TreeNodeStyled title={contract.name} key={contract._id} isLeaf={true} />
                      })} */}
                  </TreeNodeStyled>
                </TreeNodeStyled>
              )
            })}
          </DirectoryTreeStyled>
        )}
      </div>
    )
  }
}
