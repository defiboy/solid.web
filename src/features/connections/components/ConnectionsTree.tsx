import React from 'react'
import { Action, ActionCreator } from 'redux'
import { Icon } from 'antd'

import { Connection } from '../types'
import { MenuStyled, MenuItemStyled, SidebarHeader, SidebarTitle, SidebarHeaderButtons, DirectoryTreeStyled, TreeNodeStyled } from './ConnectionsTreeComponents';

interface Props {
  connections: Connection[]
  onNewConnectionClick: ActionCreator<Action>
}

interface State {
  rightClickNodeTreeItem: any
  selectedKeys: any[]
}

export class ConnectionsTree extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      rightClickNodeTreeItem: {},
      selectedKeys: []
    }
  }

  onSelect = (selectedKeys: any, info: any) => {
    this.setState({
      selectedKeys
    })
  }

  rightClickOnTree = ({ event, node }: any) => {
    const id = node.props.eventKey
    this.setState({
      rightClickNodeTreeItem: {
        pageX: event.pageX,
        pageY: event.pageY,
        id,
        categoryName: node.props.eventKey
      },
      selectedKeys: [id]
    })
  }

  getNodeTreeRightClickMenu = () => {
    const { pageX, pageY } = { ...this.state.rightClickNodeTreeItem } as any
    if (!pageX || !pageY) {
      return <div />
    }
    const menu = (
      <MenuStyled style={{ position: 'absolute', left: `${pageX}px`, top: `${pageY}px` }}>
        <MenuItemStyled data-testid="connections-tree-rightclick-menu-option-deploy" key="1">Deploy</MenuItemStyled>
        <MenuItemStyled data-testid="connections-tree-rightclick-menu-option-openconsole" key="2">Open Console</MenuItemStyled>
      </MenuStyled>
    )
    return menu
  }

  render() {
    const { connections, onNewConnectionClick } = this.props
    return (
      <div style={{ overflow: 'scroll', height: '100%' }}>
        {this.getNodeTreeRightClickMenu()}
        <SidebarHeader>
          <SidebarTitle data-testid="connections-tree-header">Connections</SidebarTitle>
          <SidebarHeaderButtons>
            <Icon type="plus" data-testid="connections-tree-plus" style={{ color: 'white', paddingRight: '0.5em' }} onClick={onNewConnectionClick} />
            <Icon type="down" data-testid="connections-tree-down" style={{ color: 'white' }} />
          </SidebarHeaderButtons>
        </SidebarHeader>
        {connections && connections.length > 0 && (
          <DirectoryTreeStyled
            onSelect={this.onSelect}
            multiple={true}
            onRightClick={this.rightClickOnTree}
            selectedKeys={this.state.selectedKeys}
            defaultExpandAll={true}
            style={{ color: 'white' }}
          >
            {connections.map(item => {
              return (
                <TreeNodeStyled
                  icon={<Icon type="database" />}
                  title={item.name}
                  key={item.url}
                  style={{ color: 'white' }}>
                  {/* <TreeNodeStyled title="Contract Definitions" key="0-0-0" style={{ color: 'white' }}>
                  </TreeNodeStyled> */}
                </TreeNodeStyled>
              )
            })}
          </DirectoryTreeStyled>
        )}
      </div>
    )
  }
}
