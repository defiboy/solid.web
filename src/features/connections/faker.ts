import { Connection } from './types'
import { generateFakeObjectId } from 'utils/fakeObjectId'
export const buildFakeConnection = (): Connection => {
    const connection: Connection = {
        name: "Connection 1",
        url: "http://localhost:8545",
        _id: generateFakeObjectId()
    }

    return connection
}

export const buildFakeConnections = (): Connection[] => {
    const connections: Connection[] = [
        buildFakeConnection(),
        {
            name: "Connection 2",
            url: "http://localhost:9000",
            _id: generateFakeObjectId()

        }]

    return connections
}
