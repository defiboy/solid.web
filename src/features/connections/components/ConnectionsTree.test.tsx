import React from 'react'
import { render, fireEvent, queryByText, findByText, getByText, wait, queryByTestId, findByTestId } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';

import { ConnectionsTree } from './ConnectionsTree'

import { Connection } from '../types'
import { buildFakeConnections } from '../faker'

import { getMouseEvent } from './__tests__/helpers/getMouseEvent';

describe('ConnectionsTree', () => {
    const onNewConnectionClickMockHandler = jest.fn()
    let connections: Connection[]

    beforeEach(() => {
        onNewConnectionClickMockHandler.mockClear()
    })

    test('that it renders all the ui elements', () => {
        const connections = buildFakeConnections()
        const text1 = connections[0].name
        const text2 = connections[1].name

        const { getByTestId, getByText, debug } = render(
            <ConnectionsTree connections={connections}
                onNewConnectionClick={onNewConnectionClickMockHandler}
            />)

        expect(getByTestId('connections-tree-header')).toBeInTheDocument()
        expect(getByTestId('connections-tree-header')).toHaveTextContent("Connections")
        expect(getByTestId('connections-tree-plus')).toBeInTheDocument()
        expect(getByTestId('connections-tree-down')).toBeInTheDocument()

        expect(getByText(text1)).toBeInTheDocument()
        expect(getByText(text2)).toBeInTheDocument()
    })

    test('that no connections are shown', () => {
        let connections = buildFakeConnections()
        const text1 = connections[0].name
        const text2 = connections[1].name
        connections = []

        const { getByTestId, queryByText } = render(
            <ConnectionsTree connections={connections}
                onNewConnectionClick={onNewConnectionClickMockHandler}
            />)

        expect(getByTestId('connections-tree-header')).toBeInTheDocument()
        expect(getByTestId('connections-tree-header')).toHaveTextContent("Connections")
        expect(getByTestId('connections-tree-plus')).toBeInTheDocument()
        expect(getByTestId('connections-tree-down')).toBeInTheDocument()

        expect(queryByText(text1)).not.toBeInTheDocument()
        expect(queryByText(text2)).not.toBeInTheDocument()
    })

    test('when right click is press in a connection, then the context menu appears', async () => {
        const connections = buildFakeConnections()
        const text1 = connections[0].name

        const { getByTestId, getByText } = render(
            <ConnectionsTree connections={connections}
                onNewConnectionClick={onNewConnectionClickMockHandler}
            />)


        const connection1Element = getByText(text1)

        fireEvent.click(connection1Element)

        const rightClick = getMouseEvent('contextmenu', {
            pageX: 250,
            pageY: 250,
            button: 2
        })

        fireEvent(connection1Element, rightClick)

        await wait(() => {
            expect(getByTestId('connections-tree-rightclick-menu-option-deploy')).toBeInTheDocument()
            expect(getByTestId('connections-tree-rightclick-menu-option-deploy')).toBeInTheDocument()
        })
    })

    test('when new connection icon is clicked', () => {
        const connections = buildFakeConnections()

        const { getByTestId } = render(
            <ConnectionsTree connections={connections}
                onNewConnectionClick={onNewConnectionClickMockHandler}
            />)


        const addConnectionIcon = getByTestId('connections-tree-plus')

        fireEvent.click(addConnectionIcon)

        expect(onNewConnectionClickMockHandler).toHaveBeenCalledTimes(1)
    })

    test('renders connections tree snapshop', async () => {
        const connections = buildFakeConnections()
        const text1 = connections[0].name

        const { getByTestId, getByText, container } = render(
            <ConnectionsTree connections={connections}
                onNewConnectionClick={onNewConnectionClickMockHandler}
            />)


        const connection1Element = getByText(text1)

        fireEvent.click(connection1Element)

        const rightClick = getMouseEvent('contextmenu', {
            pageX: 250,
            pageY: 250,
            button: 2
        })

        fireEvent(connection1Element, rightClick)

        await wait(() => {
            expect(getByTestId('connections-tree-rightclick-menu-option-deploy')).toBeInTheDocument()
            expect(getByTestId('connections-tree-rightclick-menu-option-deploy')).toBeInTheDocument()
        })

        expect(container).toMatchSnapshot()
    })
})