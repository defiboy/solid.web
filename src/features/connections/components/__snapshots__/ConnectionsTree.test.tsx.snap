// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionsTree renders connections tree snapshop 1`] = `
<div>
  <div
    style="overflow: scroll; height: 100%;"
  >
    <ul
      class="ant-menu sc-ifAKCX gJZuHy ant-menu-light ant-menu-root ant-menu-vertical"
      role="menu"
      style="position: absolute; left: 250px; top: 250px;"
    >
      <li
        class="ant-menu-item sc-EHOje jCAKrM"
        data-testid="connections-tree-rightclick-menu-option-deploy"
        role="menuitem"
      >
        Deploy
      </li>
      <li
        class="ant-menu-item sc-EHOje jCAKrM"
        data-testid="connections-tree-rightclick-menu-option-openconsole"
        role="menuitem"
      >
        Open Console
      </li>
    </ul>
    <div
      class="sc-bdVaJa cRJnvr"
    >
      <h3
        class="sc-bwzfXH boJNxd"
        data-testid="connections-tree-header"
      >
        Connections
      </h3>
      <div
        class="sc-htpNat KtFsv"
      >
        <i
          aria-label="icon: plus"
          class="anticon anticon-plus"
          data-testid="connections-tree-plus"
          style="color: white; padding-right: 0.5em;"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </i>
        <i
          aria-label="icon: down"
          class="anticon anticon-down"
          data-testid="connections-tree-down"
          style="color: white;"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </i>
      </div>
    </div>
    <ul
      class="ant-tree ant-tree-directory sc-bZQynM levuGt"
      role="tree"
      style="color: white;"
      unselectable="on"
    >
      <li
        class="sc-bxivhb iAyWlo ant-tree-treenode-switcher-close ant-tree-treenode-selected"
        role="treeitem"
        style="color: white;"
      >
        <span
          class="ant-tree-switcher ant-tree-switcher-noop"
        />
        <span
          class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-normal ant-tree-node-selected"
          title="Connection 1"
        >
          <span
            class="ant-tree-iconEle ant-tree-icon__customize"
          >
            <i
              aria-label="icon: database"
              class="anticon anticon-database"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="database"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M832 64H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V96c0-17.7-14.3-32-32-32zm-600 72h560v208H232V136zm560 480H232V408h560v208zm0 272H232V680h560v208zM304 240a40 40 0 1 0 80 0 40 40 0 1 0-80 0zm0 272a40 40 0 1 0 80 0 40 40 0 1 0-80 0zm0 272a40 40 0 1 0 80 0 40 40 0 1 0-80 0z"
                />
              </svg>
            </i>
          </span>
          <span
            class="ant-tree-title"
          >
            Connection 1
          </span>
        </span>
      </li>
      <li
        class="sc-bxivhb iAyWlo ant-tree-treenode-switcher-open"
        role="treeitem"
        style="color: white;"
      >
        <span
          class="ant-tree-switcher ant-tree-switcher-noop"
        />
        <span
          class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-normal"
          title="Connection 2"
        >
          <span
            class="ant-tree-iconEle ant-tree-icon__customize"
          >
            <i
              aria-label="icon: database"
              class="anticon anticon-database"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="database"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M832 64H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V96c0-17.7-14.3-32-32-32zm-600 72h560v208H232V136zm560 480H232V408h560v208zm0 272H232V680h560v208zM304 240a40 40 0 1 0 80 0 40 40 0 1 0-80 0zm0 272a40 40 0 1 0 80 0 40 40 0 1 0-80 0zm0 272a40 40 0 1 0 80 0 40 40 0 1 0-80 0z"
                />
              </svg>
            </i>
          </span>
          <span
            class="ant-tree-title"
          >
            Connection 2
          </span>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
